pkg/machine/qemu/command/command.go

```diff
@@ -52,10 +52,19 @@ func (q *QemuCmd) SetQmpMonitor(monitor Monitor) {
 }
 
 // SetNetwork adds a network device to the machine
-func (q *QemuCmd) SetNetwork() {
+func (q *QemuCmd) SetNetwork(vlanSocket *define.VMFile) error {
 	// Right now the mac address is hardcoded so that the host networking gives it a specific IP address.  This is
 	// why we can only run one vm at a time right now
-	*q = append(*q, "-netdev", "socket,id=vlan,fd=3", "-device", "virtio-net-pci,netdev=vlan,mac=5a:94:ef:e4:0c:ee")
+	if vlanSocket == nil {
+		return errors.New("vlanSocket is undefined")
+	}
```

Comment:
Why this nil check here and returning the error? If the caller passes nil then the caller screwed up IMO.
The extra error handling is not that great and complicates the code for no reason here.

Reply:
This could indeed be removed, especially if SetNetwork is treated as application internal API (API between same application modules, no a module as a library API), because the input is either checked at the caller level or is generated at the caller level and is supposed to be good. This code was written in a way that if the method is public, then every parameter should be handled as public method parameters and should not be trusted. This is probably my bias from JVM world, where if thing is public it is very exposed to all the stuff running in JVM.

Will remove this with a rebase onto latest main, to have CI check the changes with the latest codebase in the same turn.

Reply:
We generally do not promise any API compatibility outside of pkg/bindings.
If people start importing machine code then this is not supported by us upstream and not our problem really.